File: nodejs/250125_S5_14606.js
/**
 * https://www.acmicpc.net/problem/14606
 */
const n = Number(require('fs').readFileSync(0, 'utf-8').toString().trim());
const happiness = new Array(n+1).fill(0);
for(let i = 2; i <= n; i++) {
	const v1 = Math.trunc(i / 2);
	const v2 = i - v1;
	happiness[i] = (v1 * v2) + happiness[v1] + happiness[v2];
}
console.log(happiness[n]);

File: go/250125_S5_14606.go
/**
 * https://www.acmicpc.net/problem/14606
 */
package main

import (
	"fmt"
)

func main () {
	var n int
	fmt.Scan(&n)
	
	fun := make([]int, n+1)
	for i := 2; i <= n; i++ {
		v1 := i / 2
		v2 := i - v1
		fun[i] = (v1 * v2) + fun[v1] + fun[v2];
	}
	
	fmt.Printf("%d\n", fun[n])
}
